programs/checkout/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+            .try_borrow_mut_lamports()? -= &highest_bid.bid_price;
+        **ctx.accounts.user.try_borrow_mut_lamports()? += &highest_bid.bid_price;
+
+        Ok(())
+    }
+
+    pub fn fulfill_order(
+        ctx: Context<FulfillOrder>,
+        pubkey_to_remove: Pubkey,
+        checkout_auth_bump: u8,
+    ) -> Result<()> {
+        msg!("in remove_order pubkey: {}", &pubkey_to_remove.to_string());
+
+        let mut matched_orders = ctx.accounts.matched_orders.load_mut()?;
+        matched_orders
+            .trades
+            .remove_node_by_pubkey(pubkey_to_remove);
+
+        msg!("escrow bal: {}", ctx.accounts.escrow_voucher.amount);
+        // TODO get seeds of pda?
+        anchor_spl::token::transfer(
+            CpiContext::new_with_signer(
+                ctx.accounts.token_program.to_account_info(),
+                anchor_spl::token::Transfer {
+                    from: ctx.accounts.escrow_voucher.to_account_info(),
+                    to: ctx.accounts.order_voucher.to_account_info(),
+                    authority: ctx.accounts.checkout_auth.to_account_info(),
+                },
+                &[&[
+                    b"checkout_auth",
+                    ctx.accounts.exhibit.to_account_info().key.as_ref(),
+                    &[checkout_auth_bump],
+                ]],
+            ),
+            1,
+        )?;
+
+        Ok(())
+    }
+    // TODO which node indexes (10, 10) to search for pubkey
+}
+
+// TODO MAKE SURE STORED ACCOUNT PUBKEY CONSTRAINT WORKS
+#[derive(Accounts)]
+pub struct Initialize<'info> {
+    /// CHECK: just reading pubkey
+    pub exhibit: AccountInfo<'info>,
+
+    #[account(zero)]
+    pub matched_orders: AccountLoader<'info, MatchedOrders>,
+
+    #[account(init,
+        payer = user,
+        space = std::mem::size_of::<MatchedStorage>() + 8,
+        seeds = [b"matched_orders", exhibit.key().as_ref()], bump
+    )]
+    pub matched_storage: Account<'info, MatchedStorage>,
+
+    #[account(init,
+        payer = user,
+        space = std::mem::size_of::<BidOrders>() + 8,
+        seeds = [b"bid_orders", exhibit.key().as_ref()], bump
+    )]
+    pub bid_orders: AccountLoader<'info, BidOrders>,
+
+    #[account(init, payer = user, space = 8+std::mem::size_of::<CheckoutAuth>(), seeds=[b"checkout_auth", exhibit.key().as_ref()], bump)]
+    pub checkout_auth: Account<'info, CheckoutAuth>,
+
+    #[account(mut)]
+    pub voucher_mint: Account<'info, Mint>,
+
+    #[account(
+        init,
+        payer = user,
+        seeds = [b"escrow_voucher", checkout_auth.key().as_ref()],
+        token::mint = voucher_mint,
+        token::authority = checkout_auth,
+        bump
+    )]
+    pub escrow_voucher: Account<'info, TokenAccount>,
+
+    #[account(
+        init,
+        payer = user,
+        space = std::mem::size_of::<SolWallet>() + 8,
+        seeds = [b"escrow_sol", exhibit.key().as_ref()],
+        bump
+    )]
+    pub escrow_sol: Account<'info, SolWallet>,
```

Comment:
There's no reason to make this a struct. What you should do is seed it enough SOL for rent exemption `solana rent 0` and the call `system_instruction::transfer` to this escrow account as opposed to modifying the lamports directly